micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/web/correlationid/CorrelationIdFilter.groovy

```diff
@@ -61,16 +80,21 @@ class CorrelationIdFilter extends OncePerRequestFilter {
         return correlationId
     }
 
-    //TODO: add microservice id to corrId, so that we know where it was created
-    private String createNewCorrIdIfEmpty(String currentCorrId) {
-        if (!hasText(currentCorrId)) {
-            currentCorrId = UUID.randomUUID().toString()
-            MDC.put(CORRELATION_ID_HEADER, currentCorrId)
-            log.debug("Generating new correlationId: " + currentCorrId)
-        }
+    private String createNewCorrIdIfEmpty() {
```

Comment:
I was considering this, however it adds some subtle changes. `updateCorrelationId` logs "*Updating correlationId*" rather than "*Generating new correlationI* - which actually makes a difference. Also `CorrelationIdHolder` would be populated twice (https://github.com/4finance/micro-infra-spring/pull/197/files#diff-190e5f9988322d459cce6e0b06f16ee8R62). That being said, although it introduces some duplication, it seems like a  better idea.